azurerm/resource_arm_bastion_host.go

```diff
@@ -0,0 +1,208 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-06-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmBastionHost() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmBastionHostCreateUpdate,
+		Read:   resourceArmBastionHostRead,
+		Update: resourceArmBastionHostCreateUpdate,
+		Delete: resourceArmBastionHostDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
Could we validate this with at least `NoEmptyStrings`

Reply:
Added a validate validateAzureRMBastionHostName function for next push